src/net/sourceforge/kolmafia/request/RelayRequest.java

```diff
@@ -2830,34 +2841,37 @@ private boolean sendSpelunkyWarning(final KoLAdventure adventure) {
     }
 
     // If the player has said he doesn't care about this warning, cool
-    if (this.getFormField(CONFIRM_SPELUNKY) != null) {
+    if (this.getFormField(Confirm.SPELUNKY) != null) {
       return false;
     }
 
     // Depending on noncombat phase step and/or adventure location,
     // craft an appropriate warning
-    String message = SpelunkyRequest.spelunkyWarning(adventure, CONFIRM_SPELUNKY);
+    String message = SpelunkyRequest.spelunkyWarning(adventure, Confirm.SPELUNKY);
     if (message != null) {
       String image = SpelunkyRequest.adventureImage(adventure);
-      this.sendGeneralWarning(image != null ? image : "spelwhip.gif", message, CONFIRM_SPELUNKY);
+      this.sendGeneralWarning(image != null ? image : "spelwhip.gif", message, Confirm.SPELUNKY);
       return true;
     }
 
     return false;
   }
 
-  public void sendGeneralWarning(final String image, final String message, final String confirm) {
-    this.sendGeneralWarning(image, message, confirm, null, false);
+  public void sendGeneralWarning(final String image, final String message, final Confirm confirm) {
+    this.sendGeneralWarning(image, message, confirm.toString(), null, false);
```

Comment:
OK. These are all turning into calls on the 5-argument version of sendGeneralWarning.
Why not change the third argument of that from ```String confirm``` to ```Confirm confirm```?

Reply:
I thought the last `sendGeneralWarning` was implemented on `GenericRequest`, but I was wrong. Change made.